src/Editor/CodelensProvider.ts

```diff
@@ -19,6 +19,8 @@ import toolsAttr from './json/tools_attr.json';
 
 export class CodelensProvider implements vscode.CodeLensProvider {
   showTool: Array<string> = [];
+  eventGenerator: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
+  onDidChangeCodeLenses: vscode.Event<void> = this.eventGenerator.event;
```

Comment:
(optional) `onDidChangeCodeLenses` is not used in this change. maybe add codes that use it.

Reply:
`OnDidChangeCodeLenses` simply detects events. 
The code using `onDidChangeCodeLenses` does not appear in the code that will be added later.
Should I divide this PR into event occurrence and detection...?

Reply:
I think it is OK to add onDidChangeCodeLenses method(?) or remove this line.

Reply:
I didn't understand exactly....
`onDidChangeCodeLenses` is an event, not a method.

Additionally, I think there is a lack of explanation.
If an event occurs through `this.event Generator.fire();`, `onDidChangeCodeLenses` will detect it and re-examine the scope of the codelens to be provided.

If the text changes, re-examine the range.
However, we need to re-examine the scope to provide codelense for attributes without text changes later.
So I need `onDidChangeCodeLenses`.

Reply:
FYI https://code.visualstudio.com/api/references/vscode-api#CodeLensProvider%3CT%3E

Reply:
The vscode-api page is too huge that the page halts in my browser and I can't read the document.
Anyway, `onDidChangeCodeLenses` is just declared and not used.
I assume there will be upcoming changes that will use `onDidChangeCodeLenses` (or not used?)
For looking to this change itself, why should we put this unused variable(or something I don't know)?
This was my first comment.

And you wrote
> The code using onDidChangeCodeLenses does not appear in the code that will be added later.
Should I divide this PR into event occurrence and detection...?

So, my second comment is to add the code that uses it if you want to put `onDidChangeCodeLenses ` in this PR.

Reply:
This is a description of the `onDidChangeCodeLenses` in the `vscode.CodeLensProvider` interface.
> An optional event to signal that the code lenses from this provider have changed.

It will not be used in future codes.
It works just by assigning events to be detected to `onDidChangeCodeLenses`.

Should I add it to the next PR...?

Reply:
Sample code has `public readonly` attribute that this is readonly, will not be changed.
As I can't open the document, I assume this member is declared in `vscode.CodeLensProvider` 
and this code overrides the member, to call `this.eventGenerator.event` inside `vscode.CodeLensProvider` 
Am I right?

If so, how about modify like this? It's OK to put in this PR.
```
public readonly onDidChangeCodeLenses: vscode.Event<void> = this.eventGenerator.event;
```

Reply:
> Am I right?

You're right.
I'll change it according to your opinion.
Thank you!